src/lib/cache/gds_cache.hpp

```diff
@@ -27,8 +27,36 @@ class GDSCache : public AbstractCacheImpl<Key, Value> {
     bool operator<(const GDSCacheEntry& other) const { return priority > other.priority; }
   };
 
-  typedef GDSCacheEntry entry_t;
-  typedef typename boost::heap::fibonacci_heap<entry_t>::handle_type handle_t;
+  using entry_t = GDSCacheEntry;
+  using handle_t = typename boost::heap::fibonacci_heap<entry_t>::handle_type;
+
+  using typename AbstractCacheImpl<Key, Value>::KeyValuePair;
+  using typename AbstractCacheImpl<Key, Value>::AbstractIterator;
+  using typename AbstractCacheImpl<Key, Value>::ErasedIterator;
+
+  class Iterator : public AbstractIterator {
+   public:
+    using IteratorType = typename std::unordered_map<Key, handle_t>::iterator;
+    explicit Iterator(IteratorType p) : _map_position(p) {}
+
+   private:
+    friend class boost::iterator_core_access;
+    friend class AbstractCacheImpl<Key, Value>::ErasedIterator;
+
+    IteratorType _map_position;
```

Comment:
I don't like it tooo explicit since the information what iterator is wrapped is obvious in most cases. Nonetheless, fully agree: changed to `_wrapped_iterator` for all caches.